app/models/course/condition/assessment.rb

```diff
@@ -48,7 +48,7 @@ def self.dependent_class
   end
 
   def self.on_dependent_status_change(submission)
-    return unless submission.previous_changes.key?(:workflow_state)
+    return unless submission.changes.key?(:workflow_state)
```

Comment:
As spoken, when we upgrade to rails 5, this will be changed back to `previous_changes`.